### DISCUSSION TITLE
Perspectives like in the Eclipse IDE

### DISCUSSION THREAD

**Discussion by as-shura** [eclipse-theia/theia#12672] [Ideas]
Is there any way to use Perspectives like in the Eclipse IDE so that we have different screen configurations based on the specific functionalities we would like to separate?

**Comment by msujew** [comment]
@as-shura Not as of now. The application aligns heavily to vscode, which also doesn't feature Eclipse-IDE like perspectives. Not to say that we wouldn't support such a feature - it's just that nobody has contributed it to the framework yet.

I do believe one of the Theia contributor parties is currently investigating such a feature in Theia, see our [roadmap](https://github.com/eclipse-theia/theia/wiki/Eclipse-Theia-Roadmap) (point 1 *Enhance Workbench UX and Flexibility*). @JonasHelming any more info on that?

**Comment by JonasHelming** [comment]
Yes, several UX extensions were done, e.g. improving the tab visualization, [visual previews on hover](https://github.com/eclipse-theia/theia/pull/12648) and detachable windows.
However, there are currently no plans to implement a concept like perspectives used to be in the desktop Eclipse IDE. I do not see any issue to implement something like this. However, the question would be how exactly it fits into the current UX. In most applications based on Theia, you have the main activities on the left (File Explorer, Debugging, Git, etc.). The editor area in the middle stay the same (as it was in Eclipse desktop). The right panel often contains contextual views that are depending on the focused editor. So in a sense, switching the activity via the left panel is kind of a perspective switch in Theia. Of course this can be adapted.

I hope this helps, please let us know if you have further questions!
